gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -306,10 +319,22 @@ class UHF {
      */
     OneDM<Scalar> calculateOrthonormalBasis1DM(const Spin sigma) const {
```

Comment:
This method should return a `SpinResolvedOneDM`.

Reply:
I thought this API was meant to return either the `alpha` or `beta` component of the `OrthonormalBasis1DM` (`SpinResolved1DM`). If this is not the case, should I add an API for that as well?

Reply:
You can remove the 'old' functionality that was meant to calculate only the alpha or beta part, and replace it with the new functionality that calculates them both and encapsulates them correctly.